### PR TITLE
DRA GetStatusAlimentTimer matching

### PR DESCRIPTION
**Pull request by Xeeynamo**
Function used when the player switches status aliment. The returned value is a timer that changes based on certain parameters. It is also used by the Dark Methamorphosis spell.

I suspect the comment on `Entity.unk5E` is valid. I am refraining to fully rename it because, while that states the truth for `PLAYER`, I do not know if it is the case for all the other game entities.

### CODE REVIEW

File: src/dra/5D6C4.c [hunk 1]
```diff
@@ -144,56 +145,48 @@ void func_800FD9D4(SpellDef* spell, s32 id) {
...
         }
         if (var_v1 > 4) {
             var_v1 = 4;
         }
-        ret = ret - var_v1;
+        ret -= var_v1;
         break;
-    case 3:
-        ret = arg1 + (g_Status.statsTotal[2] * 4);
+    case 3: // Dark Methamorphosis
```
Comment (sozud) [inline src/dra/5D6C4.c:176]: Metamorphosis

File: src/dra/5D6C4.c [hunk 2]
```diff
@@ -144,56 +145,48 @@ void func_800FD9D4(SpellDef* spell, s32 id) {
...
+    s16 ret;
     s32 var_v1;
 
-    switch (arg0) {
-    case 0:
-        temp_v0 = arg1 - (g_Status.statsTotal[1] * 0x10);
-        // asm volatile("move $16, $2");
-        if (temp_v0 < 0x100) {
-            ret = 0x100;
+    switch (kind) {
```
Comment (sozud) [inline src/dra/5D6C4.c:152]: Should we have an enum for `kind`?

File: src/dra/5D6C4.c (hunk 1 above)
Comment (Xeeynamo) [reply src/dra/5D6C4.c]: Thank you, done

File: src/dra/5D6C4.c (hunk 2 above)
Comment (Xeeynamo) [reply src/dra/5D6C4.c]: Done

File: src/dra/692E8.c [hunk 3]
```diff
@@ -801,15 +801,15 @@ void func_8010FB24(void) {
...
+void func_8010FB68(void) { // Related to Dark Metamorphosis
     PLAYER.velocityY = 0;
     PLAYER.velocityX = 0;
     SetPlayerStep(Player_SpellDarkMetamorphosis);
     func_8010E3E0();
     func_8010DA48(0xBA);
     PlaySfx(NA_SE_VO_AL_DARK_METAMORPHOSIS);
     PlaySfx(NA_SE_PL_MP_GAUGE);
-    g_Player.D_80072F16 = func_800FDB18(3, 0x400);
+    g_Player.D_80072F16 = GetStatusAlimentTimer(3, 0x400);
```
Comment (sozud) [inline src/dra/692E8.c:812]: `STATUS_ALIMENT_DARK_METAMORPHOSIS, 0x400`? Feel free to go ahead and merge if you disagree, the PR is good either way.

**Review by sozud — APPROVED** [review]

File: src/dra/692E8.c (hunk 3 above)
Comment (sozud) [reply src/dra/692E8.c:812]: Oh actually ALIMENT -> AILMENT

Comment (Xeeynamo) [reply src/dra/692E8.c:812]: Good catch :P I will fix it in a bit as I am in the middle of another branch

**Review by sozud — APPROVED** [review]

**Pull request merged**